hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/physical/DetectOrderingForSlidingWindowRule.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2021 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.opt.physical;
+
+import com.hazelcast.jet.sql.impl.opt.OptUtils;
+import com.hazelcast.jet.sql.impl.opt.logical.AggregateLogicalRel;
+import com.hazelcast.jet.sql.impl.opt.logical.FullScanLogicalRel;
+import com.hazelcast.jet.sql.impl.opt.logical.ProjectLogicalRel;
+import com.hazelcast.jet.sql.impl.opt.logical.SlidingWindowLogicalRel;
+import com.hazelcast.sql.impl.QueryException;
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.plan.RelRule;
+import org.apache.calcite.plan.volcano.RelSubset;
+import org.apache.calcite.rel.RelNode;
+
+import static com.hazelcast.jet.sql.impl.opt.Conventions.LOGICAL;
+
+
+/**
+ * Detect pattern
+ * <p>
+ * - Aggregate
+ * <p>
+ * -- Project (is optional)
+ * <p>
+ * --- SlidingWindow
+ * <p>
+ * --- FullScan [stream = ture, ordered = false]
+ * <p>
+ * to throw exception with a reason
+ * `Grouping/aggregations over non-windowed, non-ordered streaming source not supported`
+ */
+public final class DetectOrderingForSlidingWindowRule extends RelRule<RelRule.Config> {
```

Comment:
We need to implement it as a trait: the add-wm rel will introduce the trait. The trait will be tied to a column. The rule will check it and fail if it cannot be fulfilled.